packages/data-schema/src/SchemaProcessor.ts

```diff
@@ -1316,6 +1386,64 @@ const mergeCustomTypeAuthRules = (
   }
 };
 
+function generateInputTypes(
+  implicitTypes: [string, any][],
+  generateInputType: boolean,
+  getRefType: ReturnType<typeof getRefTypeForSchema>,
+  authRules?: Authorization<any, any, any>[],
+  isInlineType = false,
+): string[] {
+  const generatedTypes = new Set<string>();
+
+  implicitTypes.forEach(([typeName, typeDef]) => {
+    if (isCustomType(typeDef)) {
+      const { gqlFields, implicitTypes: nestedTypes } = processFields(
+        typeName,
+        typeDef.data.fields,
+        {},
+        {},
+        getRefType,
+        undefined,
+        undefined,
+        {},
+        'dynamodb',
+        generateInputType,
+      );
+      const authString =
+        !isInlineType && authRules
+          ? mapToNativeAppSyncAuthDirectives(authRules, false).authString
+          : '';
+      const typeKeyword = generateInputType ? 'input' : 'type';
+      const customType = `${typeKeyword} ${typeName}${authString ? ` ${authString}` : ''}\n{\n  ${gqlFields.join('\n  ')}\n}`;
+      generatedTypes.add(customType);
+
+      // Process nested types
+      if (nestedTypes.length > 0) {
+        const nestedGeneratedTypes = generateInputTypes(
+          nestedTypes,
+          generateInputType,
+          getRefType,
+          authRules,
+          false,
+        );
+        nestedGeneratedTypes.forEach((type) => {
+          generatedTypes.add(type);
+        });
+      }
+    } else if (typeDef.type === 'enum') {
+      const enumDefinition = `enum ${typeName} {\n  ${typeDef.values.join('\n  ')}\n}`;
```

Comment:
Commenting out this line doesn't break tests. Either we're missing a test or this line isn't doing anything valuable.